app/util/analytics.py

```diff
@@ -114,24 +125,63 @@ def get_events(since_timestamp=None, since_id=None):
     since_timestamp = float(since_timestamp) if since_timestamp else 0.0
     since_id = int(since_id) if since_id else 0
 
-    with open(_eventlog_file, 'r') as f:
-        # todo(joey): This is inefficient since it reads the whole log file into memory (can be several megabytes).
-        # We probably want something like http://code.activestate.com/recipes/120686/
-        reversed_log_lines = reversed(f.readlines())
+    filtered_events_from_cache = get_events_from_cache(since_timestamp, since_id)
+    if len(filtered_events_from_cache) > 0:
```

Comment:
Makes sense.  I'll update my PR and change event log so that it goes to disk if all of the logs since_timestamp were not found.  Currently, we default since_timestamp to 0 when it is not specified.  I think I will change it to something like a 5 hour look back by default.  

I will default since_id to None and omit that check unless it was explicitly requested.